sweep/sweeper.go

```diff
@@ -29,13 +64,387 @@ type UtxoSweeperConfig struct {
+			if pending {
+				log.Debugf("Already pending input %v received",
+					outpoint)
+
+				// Add additional result channel to signal spend
+				// of this input.
+				pendInput.listeners =
+					append(pendInput.listeners,
+						input.resultChan)
+
+				break
+			}
+
+			// Lookup if there is already a confirmed or unconfirmed
+			// tx of ours that spends this input.
+			spendingTx, err :=
+				s.cfg.HasSpendingTx(outpoint)
+			if err != nil {
+				s.signalResultChan(input.resultChan, Result{Err: err})
+				break
+			}
+
+			// Start watching for spend of this input, either by us
+			// or the remote party. The possibility of a remote
+			// spend is there reason why just registering for conf.
+			// of our sweep tx isn't enough.
+			cancel, err := s.waitForSpend(input.input)
+			if err != nil {
+				s.signalResultChan(input.resultChan,
+					Result{Err: fmt.Errorf(
+						"cannot watch for spend: %v",
+						err)})
+
+				break
+			}
+
+			// Create a new pendingInput struct and initialize the
+			// listeners slice with the passed in result channel. If
+			// this input is offered for sweep again, the result
+			// channel will be appended to this slice.
+			pendInput = &pendingInput{
+				listeners:     []chan Result{input.resultChan},
+				ntfnRegCancel: cancel,
+			}
+
+			if spendingTx != nil {
+				spendingTxHash := spendingTx.TxHash()
+				log.Debugf("Already swept input %v received "+
+					"(tx: %v)", outpoint, spendingTxHash)
+
+				// Save spending tx to find sibling inputs in
+				// case of remote spend.
+				pendInput.txHash = spendingTxHash
+			} else {
+				log.Debugf("New input %v received", outpoint)
+
+				// If there is no spending tx yet, we include
+				// this input in the list of inputs to sweep in
+				// the next round.
+				inputToSweep[outpoint] = input.input
+			}
+
+			s.pendingInputs[outpoint] = pendInput
+
+			// Start sweep countdown timer to create opportunity for
+			// more inputs to be added.
+			if sweepCountdown == 0 {
+				sweepCountdown = s.cfg.BatchWindowSec
+			}
+
+		case spend := <-s.spendChan:
+			outpoint := spend.SpentOutPoint
+
+			var err error
+
+			// Check if spender tx hash matches the tx that we
+			// published. If not, this is a remote spend.
+			sweepTxHash := s.pendingInputs[*outpoint].txHash
+			if sweepTxHash != *spend.SpenderTxHash {
+				err = ErrRemoteSpend
+
+				// TODO: Add remaining inputs to inputsToSweep.
+			}
+
+			// If remote party is sweeping even before we published
+			// the sweep, we don't need to include it anymore.
+			delete(inputToSweep, *outpoint)
+
+			// Signal result channels sweep result.
+			s.signalResult(outpoint, Result{
+				Tx:  spend.SpendingTx,
+				Err: err,
+			})
+
+		case <-s.cfg.Ticker.Ticks():
+			log.Trace("Tick")
+			// Continuing waiting if nothing to sweep.
+			if len(inputToSweep) == 0 {
+				break
+			}
+
+			// Wait for sweep countdown
+			sweepCountdown--
+			if sweepCountdown > 0 {
+				break
```

Comment:
Alternatively we can use a `continue` here and above. IMO this more explicitly conveys the control flow, as we just want to continue to the next round of the select statement.

Reply:
I reason `break` because the case is handled. I think a matter of taste. If we decide to make it a convention from now on, I am happy to start using continue in this case.